Engine/UserInput.cs

```diff
@@ -127,7 +127,7 @@ public class CliUserInputInterface : IUserInputInterface
 
         void IUserInputInterface.RequestUserInput(object dataObject, TimeSpan timeout, bool modal)
         {
-            if(readerThread == null)
+            if(readerThread == null && Console.IsInputRedirected == false)
```

Comment:
Also, lets say i want to pipe some input into the program, there may be ways to do this and I think we should support that or it would be a somewhat difficult to explain behavior.

Alternatively, we could have an extra command line argument that can be used for when it is wanted to pipe input into the application. Just like if you do `git diff`, it behaves differently from when you do `git diff|grep something`.